### PR TITLE
cloud-controller-manager: fix missing global flags for --help

### PR DESCRIPTION
**Pull request by imjching**
**What type of PR is this?**
/kind bug

**What this PR does / why we need it**: In #67362, we introduced logical sections for the `--help` output in the `cloud-controller-manager` binary. It seems like it does not consider global flags that were registered through the `flag` package. This PR fixes that issue. We will output glog related and version flags in the "global" section, and everything else in the "generic" section.

**Which issue(s) this PR fixes**:
See #70145.

**Special notes for your reviewer**: The original PR is in #70164. The plan to is to break it down into individual PRs.

**Does this PR introduce a user-facing change?**:
```release-note
Fix missing flags in cloud-controller-manager --help.
```

/sig cli
/cc @stewart-yu @andrewsykim @dims 
/assign @deads2k

### CODE REVIEW

**Comment by jennybuckley** [comment]
cc @logicalhan

**Comment by imjching** [comment]
/hold

#68054 is trying to add in `globalflags.go` into `apiserver/pkg/util/flag/`. Will wait until that is merged in so that we could rebase and use methods from there.

File: cmd/cloud-controller-manager/app/controllermanager.go
Comment (stewart-yu) [inline cmd/cloud-controller-manager/app/controllermanager.go]: IMO, add the generic like that, no need to add function
https://github.com/kubernetes/kubernetes/blob/master/cmd/kubelet/app/server.go#L270

File: staging/src/k8s.io/apiserver/pkg/util/globalflag/globalflags.go
Comment (stewart-yu) [inline staging/src/k8s.io/apiserver/pkg/util/globalflag/globalflags.go]: and make this provate

**Review by dims — APPROVED** [review]
/lgtm

**Comment by dims** [comment]
/test pull-kubernetes-integration
/priority critical-urgent
/lgtm

Let's get this in and revisit any changes for 1.14 ( cc @andrewsykim )

**Comment by stewart-yu** [comment]
LGTM
@imjching  leave one nil suggestion, can you push the add file in https://github.com/kubernetes/kubernetes/tree/master/cmd/controller-manager/app/options directory, so we can reuse it both KCM and CCM

**Comment by imjching** [comment]
@stewart-yu Can we do what you mentioned earlier in a follow-up PR? It will be a little tricky to merge both #70216 and #70205 at the same time if both depends on the same file that we have not added in the `controller-manager` directory. (We'll probably need to rebase one on top of the other if we do that.)

**Comment by AishSundar** [comment]
@stewart-yu  do you think this PR is good to be approved for 1.13? If so can you plz approve it to enable merge?

**Comment by imjching** [comment]
I added tests for `customflags.go` so that we won't miss out any global flags easily. Could you take a look at it again? @stewart-yu @andrewsykim

**Pull request closed without merging**
